### PR TITLE
Add GitHub workflow for check Python file formatting

### PR DESCRIPTION
**Pull request by tru**
Using darker which is doing black on diffs, similar to git-clang-format.

### CODE REVIEW

**Comment by EugeneZelenko** [comment]
May be this should be part of the build as `Clang-format` in `polly`? Same could be done for Python linters (`Flake8`, `PyLint`, `MyPy`).

File: .github/workflows/pr-python-format.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+jobs:
+  python_formatting:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Fetch LLVM sources
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+          fetch-depth: 2
```
Comment (tstellar) [inline .github/workflows/pr-python-format.yml:16]: Why do we need `fetch-depth: 2` ?

Comment (tru) [reply .github/workflows/pr-python-format.yml:16]: darker is just formatting the diff for the PR - so we need the history to compute the diff from base. 2 is actually not that bad it goes pretty quickly compared to 0.

**Comment by tru** [comment]
> May be this should be part of the build as `Clang-format` in `polly`? Same could be done for Python linters (`Flake8`, `PyLint`, `MyPy`).

Not sure what you mean. I am not familiar with what polly does? We seldom "build" for python scripts.

**Comment by tru** [comment]
Ping @JDevlieghere as well since lldb has a lot of python.

**Comment by tru** [comment]
> > > May be this should be part of the build as `Clang-format` in `polly`? Same could be done for Python linters (`Flake8`, `PyLint`, `MyPy`).
> > 
> > 
> > Not sure what you mean. I am not familiar with what polly does? We seldom "build" for python scripts.
> 
> Nothing prevents to run some tools (formatters, linters) over Python source files during build. Sure, relevant rules should be added to CMake files.

That's true of course. But I don't think it will gain us anything here. It's going to be hard to do patch based formatting from CMake. We also need to install the whole toolchain and configure CMake just to run the formatters. So it adds a lot of complexity without a clear gain.

I think we might want to add darker to the pre-commit hook.

**Comment by JDevlieghere** [comment]
This is great! 

Regarding the discussion about making this part of CMake: I agree with Tobias. We don't do this for clang-format either and I don't really want a build/configure step to modify my source directory. I think a pre-commit hook or workflow makes a lot more sense.

**Comment by tru** [comment]
@tstellar any more objections or can I merge this?

**Comment by tstellar** [comment]
@tru No, this seems fine.

**Pull request merged**